### DISCUSSION TITLE
How can HA monitor communication to  RA and trigger RA to reboot at connection loss?

### DISCUSSION THREAD

**Discussion by julesmartens** [mKeRix/room-assistant#712] [Q&A]
Hi all, 
in my house  I have in several rooms raspberry zero devices installed that run RA. All connect to my Home Assistant (rpi4)via mqtt.
My problem is that the pi zero devices freeze now and then, making presence detection unreliable.
I set a cron task on every RA device, to reboot once a day.  This works fine unless RA really hangs and the cron cannot trigger a reboot.

ideally my HA continuously monitors if any RA devices lost its communicating with HA. This independent to number of Bluetooth connections.

If HA detects that one of the RA devices lost communication, HA should trigger that specific RA device to power cycle.
If HA fails to recover the communication it should  send a persistant notification so that we are aware that auto presence detection might be interrupted, such that we know we must manually deactivate the alarm before entering the house.

Q1: how can I monitor communication via mqtt?
Q2: How can I trigger a reboot via mqtt?

Thanks !

grtz Jules

**Comment by mKeRix** [comment]
The monitoring should be available out of the box in recent versions of room-assistant, as the entities should switch to unavailable on connection loss with MQTT (which is the link to HA). Rebooting isn't possible via MQTT, but you could for example use the Shell Command integration to trigger a restart via SSH.

**Reply by CobraDunn** [reply]
In my configuration, HA commonly continues to report that it's available but does not update any changes to BLE devices. It's as if one service fails but others continue to run normally.

**Comment by farrad** [comment]
Extending the [reply](https://github.com/mKeRix/room-assistant/discussions/712#discussioncomment-1045097) of @mKeRix 
Here is my solution.

I am running Docker-based Room Assistant (v2.19.0), but the script below can be also easily modified for Raspberry Pi.

On each RA node I am checking its status parsing response from `localhost:6415/status` and if an error is reported for a while I am rebooting this RA node.

Here is my `healthCheck.sh` script:
```
#!/usr/bin/env bashio

MQTT_TIMER=0
INTERVAL=5 # Seconds
MAX_TIMEOUT=15 # Seconds

while true
  do
  	sleep $INTERVAL

  	INSTANCE_STATUS=$(curl -s localhost:6415/status | jq -r '.status')
  	
  	case $INSTANCE_STATUS in
  	  ok)
		MQTT_TIMER=0
      ;;
  	  error)
    	MQTT_TIMER=$((MQTT_TIMER+INTERVAL))
      ;;
  	  *)
		bashio::log.error "Unexpected status:"
    	echo $INSTANCE_STATUS
      ;;
	esac

	if [ "$MQTT_TIMER" -gt "$MAX_TIMEOUT" ]; then
		bashio::log.info "MQTT connection lost! Restarting Room Assistant instance..."
		reboot now
	fi
done

bashio::log.error "Health Check stopped!"
```

The script is launched in a detached mode inside the RA Docker container at its start.

- `INTERVAL` sets interval in seconds between RA node status checks
- `MAX_TIMEOUT` sets the timeout in seconds after which the RA node will be rebooted.

**Reply by J0han3s** [reply]
> Extending the [reply](https://github.com/mKeRix/room-assistant/discussions/712#discussioncomment-1045097) of @mKeRix Here is my solution.
> 
> I am running Docker-based Room Assistant (v2.19.0), but the script below can be also easily modified for Raspberry Pi.
> 
> On each RA node I am checking its status parsing response from `localhost:6415/status` and if an error is reported for a while I am rebooting this RA node.
> 
> Here is my `healthCheck.sh` script:
> 
> ```
> #!/usr/bin/env bashio
> 
> MQTT_TIMER=0
> INTERVAL=5 # Seconds
> MAX_TIMEOUT=15 # Seconds
> 
> while true
>   do
>   	sleep $INTERVAL
> 
>   	INSTANCE_STATUS=$(curl -s localhost:6415/status | jq -r '.status')
>   	
>   	case $INSTANCE_STATUS in
>   	  ok)
> 		MQTT_TIMER=0
>       ;;
>   	  error)
>     	MQTT_TIMER=$((MQTT_TIMER+INTERVAL))
>       ;;
>   	  *)
> 		bashio::log.error "Unexpected status:"
>     	echo $INSTANCE_STATUS
>       ;;
> 	esac
> 
> 	if [ "$MQTT_TIMER" -gt "$MAX_TIMEOUT" ]; then
> 		bashio::log.info "MQTT connection lost! Restarting Room Assistant instance..."
> 		reboot now
> 	fi
> done
> 
> bashio::log.error "Health Check stopped!"
> ```
> 
> The script is launched in a detached mode inside the RA Docker container at its start.
> 
> * `INTERVAL` sets interval in seconds between RA node status checks
> * `MAX_TIMEOUT` sets the timeout in seconds after which the RA node will be rebooted.

Hi @farrad  - I want to implement the same fix as you. Where did you paste this code? I am using Portainer to manage my docker instances. Can this be implemented in Portainer?

Thanks
